packages/toml/src/index.ts

```diff
@@ -0,0 +1,66 @@
+import { Taplo } from '@taplo/lib'
+import type { Plugin } from 'prettier'
+
+import { defaultOptions, options } from './options'
+import type { PrettierOptions, TaploOptions } from './types'
+
+const PLUGIN_NAME = 'toml'
+
+let taplo: Taplo | null = null
+
+async function format(code: string, options: TaploOptions) {
+  if (!taplo) {
+    taplo = await Taplo.initialize()
+  }
+
+  return taplo.format(code, { options })
+}
+
+function removeBeginningTrailingNewline(code: string) {
+  code = code.replace(/^(?:\r?\n)+/, '')
+  code = code.replace(/(?:\r?\n)+$/, '')
+
+  return code
+}
+
+const TomlPlugin: Plugin<string> = {
+  languages: [
```

Comment:
https://github.com/un-ts/prettier/blob/master/scripts/languages.ts

We use a script to generate the languages field automatically from `linguist-languages` so that we would achieve better editor experience for more fields.